include/libcyphal/transport/can/transport.hpp

```diff
@@ -90,77 +175,97 @@ class TransportImpl final : public ICanTransport
 
     // MARK: IRunnable
 
-    void run(const TimePoint) override {}
+    void run(const TimePoint) override
+    {
+        std::array<cetl::byte, CANARD_MTU_MAX> payload{};
 
-private:
-    // MARK: Privates:
+        for (std::size_t media_index = 0; media_index < media_array_.size(); ++media_index)
+        {
+            CETL_DEBUG_ASSERT(media_array_[media_index] != nullptr, "Expected media interface.");
+            IMedia& media = *media_array_[media_index];
+
+            // TODO: Handle errors.
+            const auto pop_result = media.pop(payload);
+            if (const auto opt_rx_meta = cetl::get_if<cetl::optional<RxMetadata>>(&pop_result))
+            {
+                if (opt_rx_meta->has_value())
+                {
+                    const auto& rx_meta = opt_rx_meta->value();
+
+                    const auto timestamp_us =
+                        std::chrono::duration_cast<std::chrono::microseconds>(rx_meta.timestamp.time_since_epoch());
+                    const CanardFrame canard_frame{rx_meta.can_id, rx_meta.payload_size, payload.cbegin()};
+
+                    CanardRxTransfer      out_transfer{};
+                    CanardRxSubscription* out_subscription{};
+
+                    // TODO: Handle errors.
+                    const auto result = canardRxAccept(&canard_instance(),
```

Comment:
I think we need an assertion check here that we never get an argument error back from libcanard